spec/jwt_spec.rb

```diff
@@ -510,4 +510,25 @@
       end.not_to raise_error
     end
   end
+
+  context 'algorithm case insensitivity' do
+    let(:payload) { { 'a' => 1, 'b' => 'b' } }
+
+    it 'ignores algorithm casing during encode/decode' do
+      enc = JWT.encode(payload, '', 'hs256')
+      expect(JWT.decode(enc, '')).to eq([payload, { 'alg' => 'HS256'}])
+
+      enc = JWT.encode(payload, data[:rsa_private], 'ps384')
```

Comment:
The PS algorithms require OpenSSL 2.1. So until we drop support for older rubies the tests that have OpenSSL2.1 dependencies need to be separated.